packages/react-storage/src/components/StorageBrowser/Views/Controls/Table.tsx

```diff
@@ -142,24 +160,62 @@ export const TableControl: TableControl = () => {
           ))}
         </TableRow>
       </TableHead>
+
       <TableBody>
-        {rows?.map((row, rowIndex) => (
-          <TableRow key={rowIndex}>
-            {columns.map((column) => (
-              <TableData key={`${rowIndex}-${column.header}`}>
-                {/* How do we know when a row is supposed to be a button/link? */}
-                {column.key === LOCATION_BUTTON_KEY ? (
-                  <TableDataButton>
-                    <>{row[column.key]}</>
-                  </TableDataButton>
-                ) : (
-                  <>{row[column.key]}</>
-                )}
-              </TableData>
-            ))}
-          </TableRow>
-        ))}
+        {data?.map((row: U, rowIndex: number) => renderRowItem(row, rowIndex))}
       </TableBody>
     </Table>
   );
 };
+
+export const LocationsViewTable = (): JSX.Element => {
+  const [{ data, isLoading }] = useLocationsData();
+  const [, handleUpdateState] = useControl({ type: 'NAVIGATE' });
+
+  const hasLocations = !!data.result?.length;
+  const shouldRenderLocations = !hasLocations || isLoading;
+
+  const renderRowItem: RenderRowItem<LocationAccess<Permission>> = (
```

Comment:
I'm not sure if I'm able to do that because I'm using `handleUpdateState` to set up the click events for the button. Should I wrap it in a `useCallback` instead?

Reply:
That's fine for now, in the future can abstract in to a `RowItem` component